### ISSUE TITLE
[cherry-picker] [idea 💡] Extend backport sources

### ISSUE THREAD

**Issue by webknjaz** [python/cherry-picker#13]
I think with https://help.github.com/articles/checking-out-pull-requests-locally/ and https://gist.github.com/piscisaureus/3342247#gistcomment-1566295 cherry-picker could do better and even backport rebase merges and normal merge commits.

**Comment by samdoran** [comment]
Not sure if this is related, but I would _really_ love the ability to backport merge requests or a range of commits. Either pass `cherry_picker` a merge commit SHA and have it do the right thing (`-m 1` maybe?), or have it accept a range like `git cherry-pick` does with `[SHA]..[SHA]`.

**Comment by samdoran** [comment]
Actually, it looks like ranges work (I was just getting hit by a merge conflict and the `--continue` didn't to the right thing after resolving). It would still be handy to be able to pass in a merge commit, though.

**Comment by webknjaz** [comment]
I think it wouldn't be hard to detect whether some commit is a merge. OTOH it could be harder to figure out whether to pass `-m1` or `-m2` (though could probably be done by inspecting the parents and checking if it belongs to the default branch).

**Comment by samdoran** [comment]
Right. I wasn't sure about the value for `-m`. That would take a bit more work and I don't really know how complicated that could get.
